src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1700,6 +1724,20 @@ fn resolution_failure(
                         item(res),
                         variant
                     ),
+                    ResolutionFailure::MalformedGenerics => {
+                        let note = "this link has malformed generics";
+                        let help = "try adding a missing '<' or '>'";
+
+                        if let Some(span) = sp {
+                            diag.span_label(span, note);
+                            diag.span_label(span, help);
+                        } else {
+                            diag.note(note);
+                            diag.help(help);
+                        }
+
+                        return;
+                    }
```

Comment:
It looks like this:

```
warning: unresolved link to `Vec::<Box<T>::new`
  --> src/test/rustdoc/intra-doc-link-generic-params.rs:10:58
   |
10 | //! And what about something even harder? That would be [`Vec::<Box<T>::new()`].
   |                                                          ^^^^^^^^^^^^^^^^^^^^^
   |                                                          |
   |                                                          this link has malformed generics
   |                                                          try adding a missing '<' or '>'
   |
   = note: `#[warn(broken_intra_doc_links)]` on by default
```

And if the span could not be found (the `else` branch):

```
warning: unresolved link to `Vec::<Box<T>::new`
  --> src/test/rustdoc/intra-doc-link-generic-params.rs:10:58
   |
10 | //! And what about something even harder? That would be [`Vec::<Box<T>::new()`].
   |                                                          ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(broken_intra_doc_links)]` on by default
   = note: this link has malformed generics
   = help: try adding a missing '<' or '>'
```

Reply:
Should we just `panic!` if the span could not be found? In what cases would that happen?

Reply:
> Should we just panic! if the span could not be found? In what cases would that happen?

That would happen for proc-macros that generate documentation and don't pass on the `span` I think. So this definitely shouldn't panic, but a worse error message is fine (not much we can do about it).